src/v/pandaproxy/schema_registry/json.cc

```diff
@@ -992,20 +1116,24 @@ bool is_array_superset(
         // note that "additionalItems" can be defined, but it's
         // not used by validation because every element is validated against
         // "items"
-        return is_superset(
+        res.merge(is_superset(
           ctx,
           get_object_or_empty(ctx.older, older, "items"),
-          get_object_or_empty(ctx.newer, newer, "items"));
+          get_object_or_empty(ctx.newer, newer, "items"),
+          std::move(p)));
+        return res;
     }
 
     // both are tuple schemas, validation is similar to object. one side
     // effect is that the "items" key is present.
 
     // first check is for "additionalItems" compatibility, it's cheaper than the
     // rest
-    if (!is_additional_superset(
-          ctx, older, newer, additional_field_for::array)) {
-        return false;
+
+    res.merge(is_additional_superset(
+      ctx, older, newer, additional_field_for::array, p));
```

Comment:
Should this be `p / "items"`?

Reply:
In this [force-push](https://github.com/redpanda-data/redpanda/compare/b29d510aa9c4adf52e552f82711ceec20a21ebbb..6d86671a9761e69ac9b2eb90cf5d6999feca7afb) I fixed a mistake a couple of lines above this one to add "items" to the path (see the diff of the force-push and the changed test). I mistakenly thought this comment was about that line.

This line is correct as is and doesn't need "items" added to the path because it is checking the "additionalItems" field so "additionalItems" will be added to the path instead inside `is_additional_superset`. This test case corresponds to this line: https://github.com/redpanda-data/redpanda/blob/6b16be3bbb64e84e259a201c7d908741536b3cea/src/v/pandaproxy/schema_registry/test/test_json_schema.cc#L648-L655